app/scripts/modules/core/pipeline/config/stages/wait/WaitExecutionLabel.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap';
+
+import {IExecutionStageSummary} from 'core/domain/index';
+import {IExecution} from 'core/domain/IExecution';
+import {Application} from 'core/application/application.model';
+import {HoverablePopover} from 'core/presentation/HoverablePopover';
+import {ExecutionBarLabel} from 'core/pipeline/config/stages/core/ExecutionBarLabel';
+import {SkipWait} from './SkipWait';
+
+interface IState {
+  target?: any;
+}
+
+interface IProps {
+  stage: IExecutionStageSummary;
+  execution: IExecution;
+  application: Application;
+  executionMarker: boolean;
+}
+
+export class WaitExecutionLabel extends React.Component<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = {};
+  }
+
+  public render() {
+    if (!this.props.executionMarker) {
+      return (<ExecutionBarLabel {...this.props}/>);
+    }
+    const stage = this.props.stage;
+    if (stage.isRunning) {
```

Comment:
This pattern of "If some case return `<HoverablePopover>`, else return `<OverlayTrigger>`" is repeated in a lot of these labels. Is there a way this could be pulled out into a different Component?